### ISSUE TITLE
some thoughts about managing LOTS of small content pieces

### ISSUE THREAD

**Issue by lauragibbs** [elmsln/HAXcms#322]
After my mis-adventures with the Outline today (and I should have known it was not going to work; a gazillion clicks is never a good sign, ha ha), I thought I would share a couple of ideas re: managing LOTS of small content pieces, which is the kind of project I am looking at, and is perhaps relevant to other academic and non-academic projects. 

I am publishing public domain stories with HAX; I'll be published probably a couple thousand of these public domain stories each year, or even more if we can figure out what's really possible. Basically, I can expand to fill infinite story space. :-)

When I think about how to manage that, I really think about photo albums and music playlists. Flickr is a great example; if I could organize my HAX story pages the way I organize images at Flickr, that would be fabulous.
**albums**: you put items into albums; you can also order the items in an album; the same item can go in multiple albums
**collections**: you can put albums into collections; this is useful when you have a ton of albums and need to organize them
**tags**: in addition to albums/collections, you can use tags that enable searching across albums and collections
I have 8000 images at Flickr right now, mostly illustrations for stories in the public domain, and they feel totally organized to me: almost all my images are in albums, and almost all my albums are in collections, and despite the large number of items, it is all very manageable.

The Outline approach right now in the HAX theme I am using (I don't know if that Outline approach is what I would find in other themes?) is not up to the task of managing hundreds, much less thousands of items. It's too hard to arrange a ton of content in outline form, and, more importantly, the outline form prevents me from using the same item in multiple albums (playlists, books, whatever you would call the content collection).

So, those are some quick thoughts.

I know that my LOTS-of-small-content-pieces approach may not be the most common type of project you expect to see, but I hope it is one that will fit into the HAX universe of projects. :-)

**Comment by btopro** [comment]
So we haven't talked about it (at all) but there is support for fields in HAXcms, it's just only like 2 themes are using them and again... we've been keeping this kinda off radar beyond me saying it once or twice publicly. Reason being is that when we get to full on fields, there's different expectations of what this system is to be.

You bring up an interesting point on the lots of little things front. We have a lot of use-cases that sound similar as far as "I have a portfolio of work and I just want to dump these up online and be able to write a blurb about them". Which is a common Studio-esk use-case, students producing say 3-10 images that represent their body of work on an assignment / art piece, and then wanting to basically make a public exhibition with the least amount of cognitive load.

We could possibly achieve pieces of this via a bulk add page / area. Bulk add might be more of a drag and drop 50 items (images or otherwise) and before it actually uploads them, asking what you're trying to do. I can definitely see a case for a non-hierarchy based content organization system. We've been envisioning this as a new theme / design / layout, which effectively would take the "outline" and present it in a card sort / alternative manner. But, we could also just have 0 hierarchy (ala tags) and everything is flat (so the hierarchy is technically no parents). Then enhance search to allow for searching by field (tag being a field).

There's a lot in both what your asking for as well as what I'm proposing here, and by that I mean these are places we want to take the platform but I dono when human bandwidth will allow. Nikki's new forms and some UI work I've hand drawn can get part of the way to having at least the UI logic to start to get this direction. I do know though that our courses for the Fall / people we're going to be pointing to this, don't do courses like what your describing. That's not a judgement thing, it's just more of a roadmapping thing :0).

**Comment by btopro** [comment]
@lauragibbs I'm also adding an instructor backed label so that we can quantify which ideas show up here from the perspective of someone using HAX in a teaching context / that form of end-user.

**Comment by lauragibbs** [comment]
Exactly! I'm just sharing thoughts... I'm looking for pure content support, a kind of publishing system more than a course-support system. I really don't believe in "courses" per se, but I totally believe in online content for learning and sharing and connecting.

What's great is I can carry on with my assumption that Diigo is the main way that I am organizing my content, with more and more and more of my Diigo bookmarks being to HAX pages.

But if/when you can give me a way to organize my HAX pages with something other than the Outline approach (which, even if it were working, would not really be a solution for me), that will be great. In any case, I can carry on creating pages and just having them be accessible via my link bookmarks! 

One thing I will say: my guess is that faculty don't think about content in these terms is because it is just  very digital way to think of content, so different from what they are used to. But after they see an example of content-as-songs-in-playlist, I think they might really get into it. After all, they DO organize songs in playlists now. We need the same freedom/power for text content too. :-)

**Comment by btopro** [comment]
collapse ability in new outliner will help with this somewhat but we should explore adding something like this under the new site UI dashboard that hits an end point if things get reorganized or moved around.
